app/src/main/java/com/example/bankdetails/adapter/FavoritesAdapter.kt

```diff
@@ -0,0 +1,48 @@
+package com.example.bankdetails.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.bankdetails.databinding.ItemBanksBinding
+import com.example.bankdetails.model.FavoriteBank
+import com.example.bankdetails.utils.BankSelected
+
+
+class FavoritesAdapter(val bankSelectedDelegate: BankSelected) :
+    ListAdapter<FavoriteBank, FavoritesAdapter.FavouriteViewHolder>(favoriteDiffUtils) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FavouriteViewHolder {
+        val binding = ItemBanksBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return FavouriteViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: FavouriteViewHolder, position: Int) {
+        holder.bind(getItem(position))
+    }
+
+    fun getBankAt(position: Int): String {
```

Comment:
You should never have to do this. The owner of the data source is the fragment or the activity. So if you need to find a item at a position you should use that datasource.

Also this function doesnt consider the case when the position passed is beyond the bounds of the array.

Reply:
Done